### PR TITLE
Added otx18-device

### PR DESCRIPTION
**Pull request by tbrlek**
#### Summary

Added Onethinx module

#### Changes

- ...
- ...

#### Notes for Reviewers
Onethinx module is a dual core module with integrated LoRa and LoRaWAN stacks (SX126x), integrated antenna and integrated secure element. Onethinx module is not an end device, but an easy to use module "like esp32 for LoRa/LoRaWAN". Integrated stack is located on the M0+ core, while the M4 core is free for user application. It communicates with the stack through our easy-to-use Onethinx LoRa/LoRaWAN API.

I was not able to use validate for some reason.

...

#### Release Notes

- ...

### CODE REVIEW

File: vendor/index.yaml [hunk 1]
```diff
@@ -1683,3 +1683,8 @@ vendors:
   - id: elsys
     name: ElektronikSystem i Ume
     website: https://www.elsys.se/en/
+
+    - id: onethinx
```
Comment (Jaime-Trinidad) [inline vendor/index.yaml:1687]: @tbrlek onethinx already exist in index file (line 537), use that one.

**Review by Jaime-Trinidad — CHANGES_REQUESTED** [review]
files looks fine, please attend comments

**Comment by Jaime-Trinidad** [comment]
@tbrlek thanks for your submission please sign CLA

**Review by Jaime-Trinidad — APPROVED** [review]
changes look fine

File: vendor/onethinx/otx18.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+# Hardware versions (optional, use when you have revisions)
+hardwareVersions:
+  - version: '18.02'
+    numeric: 1
+
+# Firmware versions (at least one is mandatory)
+firmwareVersions:
+  - # Firmware version
+    version: '1.0'
+    numeric: 1
```
Comment (Jaime-Trinidad) [inline vendor/onethinx/otx18.yaml:13]: please add the corresponding hardware version for this firmware in this case '18.02'

File: vendor/onethinx/otx18.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+# supports LoRaWAN FUOTA via standard interfaces) and fuota other (other wireless update mechanism).
+firmwareProgramming:
+  - serial
+
+# Product and data sheet URLs (optional)
+productURL: https://www.onethinx.com/module.html
+dataSheetURL: https://onethinx.github.io/Onethinx_Documentation/DoxyDocumentation/html/intro.html
+
+# Commercial information
+resellerURLs:
```
Comment (Jaime-Trinidad) [inline vendor/onethinx/otx18.yaml:86]: in this case are resellers, region can't be World Wide

**Review by Jaime-Trinidad — CHANGES_REQUESTED** [review]
attend comments and run make validate fmt

File: vendor/onethinx/otx18.yaml (hunk 3 above)
Comment (tbrlek) [reply vendor/onethinx/otx18.yaml:86]: Currently modules can only be bought through us, and we are not "resellers" so I will delete this whole part, thanks!

File: vendor/onethinx/otx18.yaml (hunk 2 above)
Comment (tbrlek) [reply vendor/onethinx/otx18.yaml:13]: I am not sure what to add under firmware version, or how to form it. Currently out latest version is 0x000000C1 (0xC1 for short, 193 in decimal), but we plan to release C2 in the comming days (194 decimal). All firmware versions that we release work on this single hardware version that we offer. difference is in minor bug fixes. Version naming is in hexadecimal.

Older clients use older versions. Should I name all of them? All version are built based on 1.0.2B.

Comment (tbrlek) [reply vendor/onethinx/otx18.yaml:13]: Disregard, I now see I need to add hardwareVersion to firmwareVersions

**Review by Jaime-Trinidad — APPROVED** [review]

**Comment by Jaime-Trinidad** [comment]
@tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt

**Comment by tbrlek** [comment]
> @tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt

I have seen errors in my part (Onethinx). I fixed these errors, but now I am getting an error elsewhere (in Comtac).

```
Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
'bin' is not recognized as an internal or external command,
operable program or batch file.

    at ChildProcess.exithandler (child_process.js:319:12)
    at ChildProcess.emit (events.js:375:28)
    at maybeClose (internal/child_process.js:1055:16)
    at Socket.<anonymous> (internal/child_process.js:441:11)
    at Socket.emit (events.js:375:28)
    at Pipe.<anonymous> (net.js:675:12) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'bin/validate-image ./vendor/comtac/lpn-cm4-foto.png'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! lorawan-devices@1.0.0 validate: `node bin/validate.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the lorawan-devices@1.0.0 validate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Tomislav\AppData\Roaming\npm-cache\_logs\2021-06-25T12_40_45_167Z-debug.log
make: *** [validate] Error 1
```

**Comment by Jaime-Trinidad** [comment]
> > @tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt
> 
> I have seen errors in my part (Onethinx). I fixed these errors, but now I am getting an error elsewhere (in Comtac).
> 
> ```
> Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
> 'bin' is not recognized as an internal or external command,
> operable program or batch file.
> 
>     at ChildProcess.exithandler (child_process.js:319:12)
>     at ChildProcess.emit (events.js:375:28)
>     at maybeClose (internal/child_process.js:1055:16)
>     at Socket.<anonymous> (internal/child_process.js:441:11)
>     at Socket.emit (events.js:375:28)
>     at Pipe.<anonymous> (net.js:675:12) {
>   killed: false,
>   code: 1,
>   signal: null,
>   cmd: 'bin/validate-image ./vendor/comtac/lpn-cm4-foto.png'
> }
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! lorawan-devices@1.0.0 validate: `node bin/validate.js`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the lorawan-devices@1.0.0 validate script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\Tomislav\AppData\Roaming\npm-cache\_logs\2021-06-25T12_40_45_167Z-debug.log
> make: *** [validate] Error 1
> ```
> 
> Onethinx part:
> 
> ```
> onethinx: valid index
> onethinx: otx18: valid
> onethinx: otx18: EU863-870: profile onethinx/otx18-eu868-profile valid
> onethinx: otx18: US902-928: profile onethinx/otx18-us915-profile valid
> onethinx: otx18: otx18.jpg image has correct extension
> onethinx: otx18: otx18-package.jpg image has correct extension
> ```

@tbrlek update your branch, please decrease size of the image max is 2000x2000 and run make validate fmt, the error shouldn't present I run it locally without problems.

**Comment by tbrlek** [comment]
> > > @tbrlek please verify otx18-eu868-profile.yaml file it causes an error and run make validate fmt
> > 
> > 
> > I have seen errors in my part (Onethinx). I fixed these errors, but now I am getting an error elsewhere (in Comtac).
> > ```
> > Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
> > 'bin' is not recognized as an internal or external command,
> > operable program or batch file.
> > 
> >     at ChildProcess.exithandler (child_process.js:319:12)
> >     at ChildProcess.emit (events.js:375:28)
> >     at maybeClose (internal/child_process.js:1055:16)
> >     at Socket.<anonymous> (internal/child_process.js:441:11)
> >     at Socket.emit (events.js:375:28)
> >     at Pipe.<anonymous> (net.js:675:12) {
> >   killed: false,
> >   code: 1,
> >   signal: null,
> >   cmd: 'bin/validate-image ./vendor/comtac/lpn-cm4-foto.png'
> > }
> > npm ERR! code ELIFECYCLE
> > npm ERR! errno 1
> > npm ERR! lorawan-devices@1.0.0 validate: `node bin/validate.js`
> > npm ERR! Exit status 1
> > npm ERR!
> > npm ERR! Failed at the lorawan-devices@1.0.0 validate script.
> > npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> > 
> > npm ERR! A complete log of this run can be found in:
> > npm ERR!     C:\Users\Tomislav\AppData\Roaming\npm-cache\_logs\2021-06-25T12_40_45_167Z-debug.log
> > make: *** [validate] Error 1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Onethinx part:
> > ```
> > onethinx: valid index
> > onethinx: otx18: valid
> > onethinx: otx18: EU863-870: profile onethinx/otx18-eu868-profile valid
> > onethinx: otx18: US902-928: profile onethinx/otx18-us915-profile valid
> > onethinx: otx18: otx18.jpg image has correct extension
> > onethinx: otx18: otx18-package.jpg image has correct extension
> > ```
> 
> @tbrlek update your branch, please decrease size of the image max is 2000x2000 and run make validate fmt, the error shouldn't present I run it locally without problems.

@Jaime-Trinidad 
1. Updated branch. 
2. Decreased size to below 2000 x 2000. 
3. Ran make validate fmt. it still shows error at the same point for me. If it works without problems at your end, that is okay for me.
```
Error: Command failed: bin/validate-image ./vendor/comtac/lpn-cm4-foto.png
'bin' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by Jaime-Trinidad** [comment]
@tbrlek which node and npm versions are you using? `package-lock.json` must not change, please update versions and run it again.

**Comment by tbrlek** [comment]
> @tbrlek which node and npm versions are you using? `package-lock.json` must not change, please update versions and run it again.

@Jaime-Trinidad 
The readme said I need these versions: Node.js version 14.x, npm version 6.x, Go version 1.15.x. These are the versions I have:
```
node -v -> v14.17.1

npm -v -> 6.14.13

go version -> go version go1.15.13 windows/amd64
```

After I run make deps, package-lock.json changes and I get a warning.
```
C:\Onethinx_Tomislav\CoreModuleFiles\Documentation\TTNv3_Device\lorawan-devices>make deps
npm install
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
audited 76 packages in 0.931s
```
Which version do I need to have?

**Pull request merged**